plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergSmoke.java

```diff
@@ -1027,6 +1028,55 @@ public void testTruncateTextTransform()
         dropTable("test_truncate_text_transform");
     }
 
+    @Test(dataProvider = "truncateNumberTypesProvider")
+    // This particular method may or may not be @Flaky. It is annotated since the problem is generic.
+    @Flaky(issue = "https://github.com/trinodb/trino/issues/5201", match = "Failed to read footer of file: HdfsInputFile")
+    public void testTruncateNumberTransform(String dataType)
+    {
+        String table = format("test_truncate_%s_transform", dataType.toLowerCase(ENGLISH));
+        String select = "SELECT d_trunc, row_count, d.min AS d_min, d.max AS d_max, b.min AS b_min, b.max AS b_max FROM \"" + table + "$partitions\"";
```

Comment:
this should be defined after CREATE TABLE statement, otherwise it refers to columns names that haven't been defined yet